skimage/restoration/_denoise.py

```diff
@@ -56,8 +56,15 @@ def denoise_bilateral(image, win_size=5, sigma_range=None, sigma_spatial=1,
 
     """
     mode = _mode_deprecations(mode)
-    return _denoise_bilateral(image, win_size, sigma_range, sigma_spatial,
-                              bins, mode, cval)
+
+    offset = np.min(image)
+    offset_img = image - offset
+    if not offset_img.any():
+        # if all value identical, returns the same image
+        return image
+    else:
+        return _denoise_bilateral(offset_img, win_size, sigma_range, sigma_spatial,
```

Comment:
I agree that negative values should be supported, but we shouldn't mess with the range of values in the image. This should be easily possible by changing the Cython implementation accordingly.